pkg/asset/installconfig/gcp/validation.go

```diff
@@ -336,3 +337,41 @@ func validateCredentialMode(client API, ic *types.InstallConfig) field.ErrorList
 
 	return allErrs
 }
+
+func validateMarketplaceImages(client API, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	const errorMessage string = "could not find the boot image: %v"
+
+	if ic.GCP.DefaultMachinePlatform != nil && ic.GCP.DefaultMachinePlatform.OSImage.Name != "" {
+		osImage := ic.GCP.DefaultMachinePlatform.OSImage
+		_, err := client.GetImage(context.TODO(), osImage.Name, osImage.Project)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "gcp", "defaultMachinePlatform", "osImage"), osImage, fmt.Sprintf(errorMessage, err)))
+		}
```

Comment:
We should also validate the image's architecture matches that of the node. The only caveat here is that the image's arch can be UNSPECIFIED. In that case, we should probably just log a warning that the node might not work.